Realm/RLMSyncSubscription.mm

```diff
@@ -0,0 +1,483 @@
+    if (self = [super init]) {
+        _subscription = std::make_unique<realm::sync::Subscription>(subscription);
+        _subscriptionSet = subscriptionSet;
+        return self;
+    }
+    return nil;
+}
+
+- (RLMObjectId *)identifier {
+    return [[RLMObjectId alloc]initWithValue:_subscription->id()];
+}
+
+- (nullable NSString *)name {
+    const std::string_view str_view = _subscription->name();
+    std::string str = std::string(str_view);
+    const char * characters = str.c_str();
+    NSString *name = [NSString stringWithCString:characters
+                                        encoding:[NSString defaultCStringEncoding]];
+    if ([name length] == 0) {
+        return NULL;
+    }
+    return name;
+}
+
+- (NSDate *)createdAt {
+    return RLMTimestampToNSDate(_subscription->created_at());
+}
+
+- (NSDate *)updatedAt {
+    return RLMTimestampToNSDate(_subscription->updated_at());
+}
+
+- (NSString *)queryString {
+    return [NSString stringWithCString:std::string(_subscription->query_string()).c_str()
+                              encoding:[NSString defaultCStringEncoding]];
+}
+
+- (NSString *)objectClassName {
+    return [NSString stringWithCString:std::string(_subscription->object_class_name()).c_str()
+                              encoding:[NSString defaultCStringEncoding]];
+}
+
+- (void)updateSubscriptionWithClassName:(NSString *)objectClassName
+                                  where:(NSString *)predicateFormat, ... {
+    va_list args;
+    va_start(args, predicateFormat);
+    [self updateSubscriptionWithClassName:objectClassName
+                                    where:predicateFormat
+                                     args:args];
+}
+
+- (void)updateSubscriptionWithClassName:(NSString *)objectClassName
+                                  where:(NSString *)predicateFormat
+                                   args:(va_list)args {
+    [self updateSubscriptionWithClassName:objectClassName
+                                predicate:[NSPredicate predicateWithFormat:predicateFormat arguments:args]];
+    
+}
+
+- (void)updateSubscriptionWithClassName:(NSString *)objectClassName
+                              predicate:(NSPredicate *)predicate {
+    [_subscriptionSet verifyInWriteTransaction];
+    [_subscriptionSet addSubscriptionWithClassName:objectClassName
+                                  subscriptionName:self.name
+                                         predicate:predicate
+                                    updateExisting:true];
+}
+
+@end
+
+#pragma mark - SubscriptionSet
+
+@interface RLMSyncSubscriptionSet () {
+    std::unique_ptr<realm::sync::SubscriptionSet> _subscriptionSet;
+    std::unique_ptr<realm::sync::MutableSubscriptionSet> _mutableSubscriptionSet;
+}
+@end
+
+@implementation RLMSyncSubscriptionSet {
+    BOOL isInWriteTransaction;
+    RLMRealm *_realm;
+    id _strongBuffer[16];
```

Comment:
While it's admittedly a very silly thing to do, storing this here means that nested for loops on a single RLMSyncSubscriptionSet object will break.